lms/djangoapps/courseware/views/views.py

```diff
@@ -726,6 +726,12 @@ def _progress(request, course_key, student_id):
 
     # checking certificate generation configuration
     enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(student, course_key)
+
+    # We need to have id verified to show generate cert button
+    # in the case where courseMode is of Verified variety.
+    is_id_verified = SoftwareSecurePhotoVerification.user_is_verified(student) or \
+        enrollment_mode and not CourseMode.has_verified_mode(enrollment_mode)
```

Comment:
I don't think this is the right way to use the [`has_verified_mode()`](https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py#L415). It's meant to accept a dictionary mapping course mode slugs to modes. It happens to work here because you're passing it a string.